### PR TITLE
Better control over flow replicas

### PR DESCRIPTION
**Pull request by istalker2**
Now it is possible to explicitly specify replicaSet name.
Different flows with the same replicaSet will share replicas.

Also now it is possible to use $AC_FLOW_NAME
variables in resource definitions

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/237)

### CODE REVIEW

**Comment by nebril** [comment]
Looks good overall, but `ReplicaSet` name is the same as core k8s data structure and will cause confusion. Please consider changing the name.

**Comment by pigmej** [comment]
+1 for changing name as @nebril suggests.

**Comment by istalker2** [comment]
fixed

File: pkg/scheduler/dependency_graph.go [hunk 1]
```diff
@@ -381,11 +383,15 @@ func (rl *SortableReplicaList) Swap(i, j int) {
...
+	name := flow.ReplicaSpace
+	if name == "" {
+		if options.FlowInstanceName != "" {
+			name = options.FlowInstanceName
+		} else {
+			name = flow.Name
+		}
 	}
-	label := labels.Set{"flow": name}
+	label := labels.Set{"replicaspace": name}
```
Comment (jellonek) [inline pkg/scheduler/dependency_graph.go]: Above lines could be extracted as `getNameAndLabel`.

**Comment by istalker2** [comment]
updated

**Comment by jellonek** [comment]
Looking only on last commit - LGTM.

**Pull request merged**